cedar-lean/Cedar/Thm/Data/Set.lean

```diff
@@ -179,6 +199,14 @@ theorem make_wf [LT α] [DecidableLT α] [StrictLT α] (xs : List α) :
 := by
   simp only [WellFormed, make, toList, elts, List.canonicalize_idempotent]
 
+theorem make_sorted {α} [LT α] [DecidableLT α] [StrictLT α] {xs : List α} :
+  xs.Sorted → Set.make xs = Set.mk xs
+:= by
+  intro h
+  have hid : (fun x => x) = @id α := by
+    apply funext ; simp only [id_eq, implies_true]
```

Comment:
Is this not true by definitional equality?  Does the `simp only` below fail if you don't include `hid` as a lemma?

Reply:
Surprisingly, this is necessary, and `simp only` does indeed fail without it.

Reply:
Weird